src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -914,5 +925,16 @@ export class TransactionApi implements ITransactionApi {
       messageHash: args.messageHash,
     });
     await this.cacheService.deleteByKey(key);
+    await this._setInvalidationTimeForKey(key);
+  }
+
+  private async _setInvalidationTimeForKey(key: string): Promise<void> {
```

Comment:
Note: instead of having this function here, we could opt for moving this logic to the `CacheService` itself. That would in fact simplify things, but would also write the `invalidationTimeMs` entry on each deletion, which could be a problem if we want to trigger a deletion and avoid doing that.

Reply:
I am in favour of moving this as there is a lot of repetition and setting it is also easily forgotten. We could add a flag argument or a different method for setting the `invalidationTimeMs` alongside deletion. We can discuss this further and implement it later though.

Reply:
I've moved this logic to the `CacheService` implementations on a single commit ([1d4da4c](https://github.com/safe-global/safe-client-gateway/pull/970/commits/1d4da4c0eabcd9f200cb0e9e813d8535421aabea)). But I have to say I don't have a strong opinion regarding this, there are pros and cons of each option. @fmrsabino what's your take on this? 🙂

Reply:
If the end result is that we create a new key with the invalidation time then I would say that this should be ok and have it created all the time to simplify things.

I'd argue that the value on having this key every time outweighs the branching logic that we would need to have on each feature that we want to cache (as it stands right now of course. This can change in the future).

This is also related with my comment [here](https://github.com/safe-global/safe-client-gateway/pull/970#discussion_r1435047173)